pytorch_toolkit/ote/ote/tests/test_case.py

```diff
@@ -0,0 +1,195 @@
+                f' --resume-from snapshot.pth'
+                f' --save-checkpoints-to {self.template_folder}'
+                f' --gpu-num 1'
+                f' --batch-size 1'
+                f' --epochs {self.total_epochs}'
+                f' | tee {log_file}')
+
+            self.assertTrue(os.path.exists(os.path.join(self.template_folder, 'latest.pth')))
+
+        @unittest.skipUnless(torch.cuda.is_available(), 'No GPU found')
+        def test_evaluation_on_gpu(self):
+            self.do_evaluation(on_gpu=True)
+
+        def test_evaluation_on_cpu(self):
+            skip_if_cpu_is_not_supported(self.template_file)
+            self.do_evaluation(on_gpu=False)
+
+        @unittest.skipUnless(torch.cuda.is_available(), 'No GPU found')
+        def test_finetuning_on_gpu(self):
+            self.do_finetuning(on_gpu=True)
+
+        def test_finetuning_on_cpu(self):
+            skip_if_cpu_is_not_supported(self.template_file)
+            self.do_finetuning(on_gpu=False)
+            
+
+    return TestCaseOteApi
+
+
+def create_export_test_case(domain_name, problem_name, model_name, ann_file, img_root, metric_keys, expected_outputs_dir):
+    class ExportTestCase(unittest.TestCase):
+
+        @classmethod
+        def setUpClass(cls):
+            cls.templates_folder = os.environ['MODEL_TEMPLATES']
+            cls.template_folder = os.path.join(cls.templates_folder, domain_name, problem_name, model_name)
+            skip_non_instantiated_template_if_its_allowed(cls.template_folder, problem_name, model_name)
+            cls.template_file = os.path.join(cls.template_folder, 'template.yaml')
+            cls.ann_file = ann_file
+            cls.img_root = img_root
+            cls.dependencies = get_dependencies(cls.template_file)
+            cls.test_export_thr = 0.031
+
+            expected_outputs_json = f'{expected_outputs_dir}/{problem_name}/{model_name}.json'
+
+            with open(expected_outputs_json) as read_file:
+                cls.expected_outputs = json.load(read_file)
+
+            download_snapshot_if_not_yet(cls.template_file, cls.template_folder)
+
+        def do_evaluation(self, export_dir):
+            run_through_shell(
+                f'cd {os.path.dirname(self.template_file)};'
+                f'python eval.py'
+                f' --test-ann-files {ann_file}'
+                f' --test-data-roots {img_root}'
+                f' --load-weights {os.path.join(export_dir, "model.bin")}'
+                f' --save-metrics-to {os.path.join(export_dir, "metrics.yaml")}'
+            )
+
+            with open(os.path.join(export_dir, "metrics.yaml")) as read_file:
+                content = yaml.load(read_file, yaml.SafeLoader)
+
+            for metric_key in metric_keys:
+                value = [metrics['value'] for metrics in content['metrics'] if metrics['key'] == metric_key][0]
+                self.assertGreater(value, self.expected_outputs[metric_key] - self.test_export_thr)
```

Comment:
Do we need to divide `value` by 100 here like in the same method in `TestCaseOteApi` class? And in general, do we need to multiply by 100 the collected metrics [here](https://github.com/openvinotoolkit/training_extensions/blob/feature/ik/text_spotting2/pytorch_toolkit/ote/ote/metrics/detection/common.py#L43)?

Reply:
Thanks, I guess this is just matter of convention. For now all metrics such as `bbox` ap, `segm` ap, `f1` score and `word_spotting` are measured in percent, and from my perspective it is convenient to visualize such values in Web UI, pointing units(%) as well. So I have multiplied expected outputs by factor of 100 in order to avoid extra conversion errors as in this case.

Reply:
OK. Are tests passed now?

Reply:
Running locally, had to change expected outputs for cascade-rcnn models and increase threshold a bit.

@druzhkov-paul @ygnn123 
https://github.com/openvinotoolkit/training_extensions/pull/383/commits/8c39e0b14a3146bbe8f6c5b31ae90b17597eed58

Reply:
passed